Failing to create crossenv environment for python3.10

**Build-Python:** /opt/python/cp310-cp310/bin/python

**Cross-Python:** /opt/_internal/xc/xcpython-3.10.0b3/bin/python3

**Error encountered:** AssertionError: sysconfig was set up prior to patching?

**Logs:**
> [root:/work] # /opt/python/cp310-cp310/bin/python -m crossenv /opt/_internal/xc/xcpython-3.10.0b3/bin/python3 env
WARNING: Unable to symlink '/opt/python/cp310-cp310/bin/python' to '/work/env/cross/bin/python'
Traceback (most recent call last):
  File "/opt/python/cp310-cp310/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/python/cp310-cp310/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/work/env/build/lib/python3.10/site-packages/pip/__main__.py", line 29, in <module>
    from pip._internal.cli.main import main as _main
  File "/work/env/build/lib/python3.10/site-packages/pip/_internal/cli/main.py", line 9, in <module>
    from pip._internal.cli.autocompletion import autocomplete
  File "/work/env/build/lib/python3.10/site-packages/pip/_internal/cli/autocompletion.py", line 10, in <module>
    from pip._internal.cli.main_parser import create_main_parser
  File "/work/env/build/lib/python3.10/site-packages/pip/_internal/cli/main_parser.py", line 8, in <module>
    from pip._internal.cli import cmdoptions
  File "/work/env/build/lib/python3.10/site-packages/pip/_internal/cli/cmdoptions.py", line 23, in <module>
    from pip._internal.cli.parser import ConfigOptionParser
  File "/work/env/build/lib/python3.10/site-packages/pip/_internal/cli/parser.py", line 12, in <module>
    from pip._internal.configuration import Configuration, ConfigurationError
  File "/work/env/build/lib/python3.10/site-packages/pip/_internal/configuration.py", line 27, in <module>
    from pip._internal.utils.misc import ensure_dir, enum
  File "/work/env/build/lib/python3.10/site-packages/pip/_internal/utils/misc.py", line 42, in <module>
    from pip._internal.locations import get_major_minor_version, site_packages, user_site
  File "/work/env/build/lib/python3.10/site-packages/pip/_internal/locations/__init__.py", line 9, in <module>
    from . import _distutils, _sysconfig
  File "/work/env/build/lib/python3.10/site-packages/pip/_internal/locations/_distutils.py", line 9, in <module>
    from distutils.command.install import SCHEME_KEYS
  File "/opt/python/cp310-cp310/lib/python3.10/distutils/command/install.py", line 13, in <module>
    from distutils.sysconfig import get_config_vars
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "/work/env/lib/site.py", line 68, in exec_module
    _patch_module(module, self.patch)
  File "/work/env/lib/site.py", line 55, in _patch_module
    exec(src, module.__dict__, module.__dict__)
  File "<string>", line 21, in <module>
AssertionError: sysconfig was set up prior to patching?
ERROR: Command '['/work/env/cross/bin/python', '-m', 'pip', '--disable-pip-version-check', 'install', '--ignore-installed', '--prefix=/work/env/cross', 'pip==21.1.1', 'setuptools==56.0.0']' returned non-zero exit status 1.
[root:/work] 3s 1 #

--- a/crossenv/scripts/distutils-sysconfig-patch.py.tmpl
+++ b/crossenv/scripts/distutils-sysconfig-patch.py.tmpl
@@ -1,23 +1,27 @@
 # Patch the things that depend on os.environ or sys
 # This is very similar to the sysconfig patch
 
-project_base = {{repr(self.host_project_base)}}
+# Only patch if get_config_vars was implemented in this module. Python 3.10
+# merged the implementations by importing from sysconfig, so we don't need to
+# patch twice.
+if get_config_vars.__module__ == __name__:
+    project_base = {{repr(self.host_project_base)}}
 
-def get_makefile_filename():
-    return {{repr(self.host_makefile)}}
+    def get_makefile_filename():
+        return {{repr(self.host_makefile)}}
 
-__real_init_posix = _init_posix
-def _init_posix():
-    old = os.environ.get('_PYTHON_SYSCONFIGDATA_NAME')
-    os.environ['_PYTHON_SYSCONFIGDATA_NAME'] = {{repr(sysconfig_name)}}
-    try:
-        return __real_init_posix()
-    finally:
-        if old is None:
-            del os.environ['_PYTHON_SYSCONFIGDATA_NAME']
-        else:
-            os.environ['_PYTHON_SYSCONFIGDATA_NAME'] = old
+    __real_init_posix = _init_posix
+    def _init_posix():
+        old = os.environ.get('_PYTHON_SYSCONFIGDATA_NAME')
+        os.environ['_PYTHON_SYSCONFIGDATA_NAME'] = {{repr(sysconfig_name)}}
+        try:
+            return __real_init_posix()
+        finally:
+            if old is None:
+                del os.environ['_PYTHON_SYSCONFIGDATA_NAME']
+            else:
+                os.environ['_PYTHON_SYSCONFIGDATA_NAME'] = old
 
-assert _config_vars is None, "sysconfig was set up prior to patching?"
+    assert _config_vars is None, "distutils.sysconfig was set up prior to patching?"
 
 #vi: ft=python

